pkg/registry/storage.go

```diff
@@ -0,0 +1,18 @@
+package registry
+
+import (
+	"context"
+
+	orasregistry "oras.land/oras-go/v2/registry"
+
+	releasev1 "github.com/aws/eks-anywhere/release/api/v1alpha1"
+)
+
+// StorageClient interface for general image storage client.
+type StorageClient interface {
```

Comment:
could we just define this where it's used? or do you envision this being needed as a shared interface for something else?

Reply:
I was planning on using this interface for a file storage client, not just a registry client. The file client would read/write from a directory or tarball.

Reply:
The file storage client already exists in oras, but it is not completely implemented. They are planning on finishing it up in January.

Reply:
I see. I don't fully understand it but I'm cool moving forward with this and we can revisit if necessary when we add the file storage client.

Reply:
The file storage client would be for download/import so we could have those in OCI format as well, but future thing.